docs/spark/tutorials/get-started-macos-instructions.md

```diff
@@ -0,0 +1,128 @@
+  - Install using the installer and verify you are able to run `java` from your command-line
+- Download and install **[Apache Spark 2.4.4](https://archive.apache.org/dist/spark/spark-2.4.4/spark-2.4.4-bin-hadoop2.7.tgz)**:
+  - Add the necessary environment variables SPARK_HOME e.g., `~/bin/spark-2.4.4-bin-hadoop2.7/`
+  
+    ```bash
+    export SPARK_HOME=~/bin/spark-2.4.4-bin-hadoop2.7/
+    export PATH="$SPARK_HOME/bin:$PATH"
+    source ~/.bashrc
+    ```
+    
+- Download and install **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release:
+  - Select a **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release from .NET for Apache Spark GitHub Releases page and download into your local machine (e.g., `/bin/Microsoft.Spark.Worker/`).
+  - **IMPORTANT** Create a new environment variable using ```export DOTNET_WORKER_DIR <your_path>``` and set it to the directory where you downloaded and extracted the Microsoft.Spark.Worker (e.g., `/bin/Microsoft.Spark.Worker/`).
+
+## Authoring a .NET for Apache Spark App
+
+1. Use the .NET Core CLI to create a console application.
+
+    ```dotnetcli
+    dotnet new console -o HelloSpark
+    ```
+    
+1. Install the `Microsoft.Spark` NuGet package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package)
+    
+    ```
+    cd HelloSpark
+    dotnet add package Microsoft.Spark
+    ```
+    
+1. Replace the contents of the *Program.cs* file with the following code:
+    
+    ```csharp
+    using Microsoft.Spark.Sql;
+
+    namespace HelloSpark
+    {
+        class Program
+        {
+            static void Main(string[] args)
+            {
+                var spark = SparkSession.Builder().GetOrCreate();
+                DataFrame df = spark.Read().Json("people.json");
+                df.Show();
+            }
+        }
+    }
+    ```
+    
+1. Use the .NET Core CLI to build the application:
+    
+    ```dotnetcli
+    dotnet build
+    ```
+
+## Running your .NET for Apache Spark App
+
+1. Open your terminal and navigate into your app folder:
+    
+    ```bash
+    cd <your-app-output-directory>
+    ```
+    
+1. Create *people.json* with the following content:
+    
+    ```json
+    { "name" : "Michael" }
+    { "name" : "Andy", "age" : 30 }
+    { "name" : "Justin", "age" : 19 }
+    ```
+    
+1. Run your app
+    
+    ```bash
+    spark-submit \
+    --class org.apache.spark.deploy.dotnet.DotnetRunner \
+    --master local \
+    microsoft-spark-2.4.x-<version>.jar \
+    dotnet HelloSpark.dll 
+    ```
+    
+    > [!NOTE]
+    > This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`, otherwise, you would have to use the full path (e.g., `~/spark/bin/spark-submit`).
```

Comment:
```suggestion
    > This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`. Otherwise, you would have to use the full path (for example, `~/spark/bin/spark-submit`).
```